src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -805,6 +801,11 @@ export function getInputType(
     }
 
     function getInputTypeForUnion(union: Union): InputUnionType | InputType {
+        var clientType = getClientType(context, union);
```

Comment:
Does `getClientType` handle all kinds of Union, and convert it to correct type such as enum, union or nullable type? If so, maybe we just need to call `getClientType` and remove duplicate handling below.

Reply:
It is expected to. But currently it is only prepared for extensible enum, so extensible enum is the scope of this PR. When it is prepared for other types, we would integrate it.